datajunction-server/datajunction_server/internal/nodes.py

```diff
@@ -1388,6 +1389,29 @@ async def column_lineage(
     return lineage_column
 
 
+async def derive_sql_column(
+    cube_element: CubeElementMetadata,
+) -> ColumnOutput:
+    """
+    Derives the column name in the generated Cube SQL based on the CubeElement
+    """
+    query_column_name = None
+    if cube_element.type == "metric":
```

Comment:
Is this the best way to check for type "metric" ?

Reply:
I think so...I don't like to match on strings either but at this point in the logic we're just working with the pydantic model that just has metadata (dicts, strings, and ints).